library/core/src/slice/sort.rs

```diff
@@ -389,11 +405,20 @@ where
 
         if start_l == end_l {
             // All out-of-order elements in the left block were moved. Move to the next block.
+
+            // SAFETY: if `!is_done` then the slice width is guaranteed to be at least `2*BLOCK` wide. There
+            // are at most `BLOCK` elements in `offsets_l` because of its size, so the `offset` operation is
+            // safe. Otherwise, the debug assertions above guarantee that `width(l, r) == block_l + block_r`,
+            // namely, that the block sizes have been adjusted to account for the smaller number of remaining
+            // elements.
             l = unsafe { l.offset(block_l as isize) };
         }
 
         if start_r == end_r {
             // All out-of-order elements in the right block were moved. Move to the previous block.
+
+            // SAFETY: Same argument as above. Either this is a full block `2*BLOCK`-wide, or `block_r` has
```

Comment:
Just a small nit: if can be unclear if this refers to the previous comment or the previous Safety comment. I don't have a link but you can "name" the referred comment something like this:

```
// a-somewhat-logical-name
// SAFETY: expl

...

// SAFETY: expl
// also refer to comment [a-somewhat-logical-name] for more reasons
```